### ISSUE TITLE
Can't create new edge (js)

### ISSUE THREAD

**Issue by nobilik** [bcakmakoglu/vue-flow#50]
Hi! At first thanks for your work! 
Can't understand what I'm doing wrong...
```
    const onConnect = (params) => {
      console.log("on connect", params);
      elements.value = addEdge(params, elements.value);
    };
```
and the console output is:
```
source: "3"​
sourceHandle: null​
target: "4"​
targetHandle: ""
```
`Uncaught TypeError: t is undefined`

my custom node: 
```
<template>
  <Handle
    type="target"
    :position="Position.Left"
    :style="{
      background: '#fff',
      border: '1px solid',
      'border-color': getCssVar('warning'),
      width: '10px',
      height: '10px',
    }"
  />
  <q-card>
    <q-card-section class="bg-warning text-white text-h6">
      {{ node.label }}
    </q-card-section>
    <q-card-section>
      <q-input label="Sec" outlined="" v-model="node.value" />
    </q-card-section>
  </q-card>
  <Handle
    type="source"
    :position="Position.Right"
    :style="{
      background: '#fff',
      border: '1px solid',
      'border-color': getCssVar('warning'),
      width: '10px',
      height: '10px',
    }"
  />
</template>

<script>
import { defineComponent, ref } from "vue";
import { Handle, Position } from "@braks/vue-flow";

import { getCssVar } from "quasar";

export default defineComponent({
  name: "MessageNode",
  components: {
    Handle,
  },
  props: ["data"],
  setup(props) {
    const node = ref(props.data);
    return {
      getCssVar,
      Position,
      node,
    };
  },
});
</script>

```

**Comment by bcakmakoglu** [comment]
Hi and thanks!

Are you using version 0.3.x? Or the beta (0.4.0-x)?

**Comment by nobilik** [comment]
@bcakmakoglu sorry 0.4.0-12

**Comment by bcakmakoglu** [comment]
The label prop has been moved from `data` so you might be getting the error because you're trying to access label from `node.data.label`.
So it's not `node.data.label` anymore but just `node.label`.
This is of course only true, if you're not manually passing the label in your data object 😄 

If you're on 0.4.0-x and using composition-api I recommend *not* using the `addEdge` utility but instead use the `addEdges` utility that is provided by `useVueFlow`:
```vue
<script setup>
import { useVueFlow, VueFlow } from '@braks/vue-flow'

const elements = ref([...]) // elements omitted for simplicity

const { addEdges, onConnect } = useVueFlow()
onConnect((params) => addEdges([params]))
</script>
<template>
	<VueFlow v-model="elements" />
</template>
```

**Comment by bcakmakoglu** [comment]
Can you explain that more?
You create an edge with `addEdges` - then remove the edge - how? just removing from your elements? - then add _the same_ edge again? Or are you just adding another one with `addEdges`?

**Comment by bcakmakoglu** [comment]
I just deployed `0.4.0-14` - can you update your version and see if the issue persists? Just added a fix that might be related to the issue here as I can't seem to reproduce it using the steps you described.
Otherwise I'll try to set up a Quasar project and see if I can reproduce it that way.

**Comment by nobilik** [comment]
Sorry for yesterday I was falling asleep. So, the problem was that I use <template> with multiple blocks <handle><card><handle> When I wrap everything into <div> the error gone. But :smile: now I have this warning `Failed setting prop "parentNode" on <div>: value undefined is invalid. TypeError: setting getter-only property "parentNode"`

**Comment by bcakmakoglu** [comment]
I checked your code and there are some issues in it, you're importing ts-types in a js-vue file and mixing 0.3.x API with 0.4.0-x API.
[Here's a sandbox link](https://codesandbox.io/s/vue-flow-playground-forked-z20qcg?file=/src/App.vue) using your code (though without quasar) that throws no errors/warnings.

First thing to note is, if you're using composition, you can bind all events in your setup function, so no need to use `@event`.
If you got some static options to pass, you can also just pass them to `useVueFlow`, so for example if you want to set the `connectionMode` do `useVueFlow({ connectionMode: ConnectionMode.Loose })`.
Second, `nodeTypes` and `slots` should be **either or**. 

With version 0.4.0-x you pass either a node-type as an Object (changed from array in 0.3.x), i.e. 
```js
import { markRaw } from 'vue'
import CustomNode from './CustomNode.vue'

// mark it as raw to prevent vue from making the component definition a reactive object (otherwise it will throw a warning)
const nodeTypes = { custom: markRaw(CustomNode) }
```

**or** you use the slot
```vue
<template>
	<VueFlow>
		<template #node-custom="props">
			<CustomNode v-bind="props" />
		</template>
	<VueFlow />
</template>
```

I know the docs aren't exactly clear on that at the moment 😅 
But the sandbox should give you an idea on how to implement it correctly.

Also a side-note, if you're using composition-api anyway I highly recommend using `<script setup>`-syntax.
You can find the docs on that [here](https://vuejs.org/api/sfc-script-setup.html).

**Comment by nobilik** [comment]
As far as I understand `inheritAttrs: false,` removes the warning. No warnings. Thanks. But take a look for _Edge created between one node handles on handle doubleclick even on not custom node_ You can see it even in your code snippet

**Comment by bcakmakoglu** [comment]
`inheritAttrs: false` just prevents props that you do not expect from being bound - which causes a warning usually.
You can remove `inheritAttrs` (which is true by default) and instead of using `v-bind="props"` bind your props seperately, i.e. 
`<CustomNode :data="props.data" :label="props.label" />` so you avoid binding props your component does not expect.

The second part of your question I don't understand, 
You have to explain that a little more detailed because I can't follow what's wrong 😕 
You create an edge and it handles a doubleclick when it shouldn't? Or what exactly is going wrong?

**Comment by bcakmakoglu** [comment]
Another note: 
`Failed setting prop "parentNode" on <div>: value undefined is invalid. TypeError: setting getter-only property "parentNode"`

This error currently happens when you use a custom type for a node but you do not properly pass a slot or component for that type thus in your DOM there'll be an element with the <custom>-tag, which is not a valid tag to bind props to, hence the error.
I fixed that issue and gave a proper warning and nodes that cannot be resolved will fall back to the default node in the future (as it was intended in the first place).

**Comment by nobilik** [comment]
> The second part of your question I don't understand, You have to explain that a little more detailed because I can't follow what's wrong confused You create an edge and it handles a doubleclick when it shouldn't? Or what exactly is going wrong?
follow [your link](https://codesandbox.io/s/vue-flow-playground-forked-z20qcg?file=/src/App.vue) and doubleclick one of the handles on DelayNode for example. You will see an edge created between DelayNode handles

**Comment by bcakmakoglu** [comment]
Ahh, now I see what you mean. Yeah that should obviously be prevented, good catch, thanks 😄 
I will fix that asap.

**Comment by bcakmakoglu** [comment]
Issue is fixed with `0.4.0-17`.
